constellations/gemini-admin/src/user/userInput.js

```diff
@@ -0,0 +1,92 @@
+"use strict";
+require("./userInput.sass");
+import React from "react";
+import { InputText, emptyValidator } from "../utils/inputText.js";
+
+export class UserInput extends React.Component {
+    render = () => {
+        const handleChange = this.props.handleChange;
+
+        return (
+            <div id="view-user-inputs">
+                <InputText
+                    label="First Name"
+                    description="Enter your first name"
+                    required={true}
+                    value={this.props.firstName}
+                    onChangeCallback={(e) => handleChange(e, "firstName")}
+                    validators={[emptyValidator("name")]}
+                />
+
+                <InputText
+                    label="Middle Name"
+                    description="Enter your middle name if applicable"
+                    value={this.props.middleName}
+                    onChangeCallback={(e) => handleChange(e, "middleName")}
+                />
+
+                <InputText
+                    label="Last Name"
+                    description="Enter your last name"
+                    required={true}
+                    value={this.props.lastName}
+                    onChangeCallback={(e) => handleChange(e, "lastName")}
+                    validators={[emptyValidator("name")]}
+                />
+
+                <InputText
+                    label="Email"
+                    description="Enter your email address"
+                    required={true}
+                    value={this.props.email}
+                    onChangeCallback={(e) => handleChange(e, "email")}
+                    validators={[
+                        emptyValidator("email"),
+                        {
+                            validate: (email) =>
+                                /^[^( @)]+@[^( @)]+\.[^( @)]+$/.test(email),
+                            message: "Invalid email address",
+                        },
+                    ]}
+                />
+
+                <InputText
+                    label="Phone"
+                    description="Enter your phone number"
+                    required={true}
+                    value={this.props.phone}
+                    onChangeCallback={(e) => handleChange(e, "phone")}
+                    validators={[
+                        emptyValidator("phone number"),
+                        {
+                            validate: (num) => /^[\d\s+.()/-]{3,}$/.test(num),
+                            message: "Invalid phone number",
```

Comment:
Same with here, i'd prefer to not have regex matching in the middle of some HTML, so let's hide it away in a `validatePhoneNumber` function. If we use a utility function, even better because then other parts of the application can use it! And it'll be guaranteed to be the same logic in diff parts of the app!